python-package/xgboost/sklearn.py

```diff
@@ -476,6 +466,9 @@ def fit(self, X, y, sample_weight=None, base_margin=None,
 
                 [xgb.callback.reset_learning_rate(custom_rates)]
         """
+        self.missing = self.missing if self.missing is not None else np.nan
```

Comment:
It is not allowed to change public attributes (without trailing or leading `_`):
https://github.com/scikit-learn/scikit-learn/blob/b194674c42d54b26137a456c510c5fdba1ba23e0/sklearn/utils/estimator_checks.py#L949-L955
Instead, `self.missing_` should be created in `fit()` and used later.

Reply:
You are right, I think it's fixed now.